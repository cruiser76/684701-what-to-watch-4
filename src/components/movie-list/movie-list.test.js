import React from 'react';
import renderer from 'react-test-renderer';

import MovieList from './movie-list.jsx';

const movies = [
  {
    img: {
      src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      posterSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      bgSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
    },
    brief: {
      title: `Fantastic Beasts: The Crimes of Grindelwald`,
      genre: `Drama`,
      year: `2000`,
      score: `9`,
      level: `Good`,
      filmLink: ``
    },
    link: `movie-page.html`,
    key: `fantastic-beasts-the-crimes-of-grindelwald`,
  }
];

const props = {
  movies,
  genresList: [`All genres`],
  activeGenre: `All genres`,
  moviesList: movies,
  isMoreBtnShow: true,
  numberMoviesInList: 8,
  onCardClick: () => {},
  onMoreBtnClick: () => {},
  onGenreClick: () => {},
  renderActiveCard: () => {},
};

it(`Movie-list is render correctly`, () => {
  const tree = renderer
    .create(
        <MovieList
          {...props}
        />, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
