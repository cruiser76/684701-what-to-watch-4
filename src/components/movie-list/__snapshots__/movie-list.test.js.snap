// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie-list is render correctly 1`] = `
<div
  className="catalog__movies-list"
/>
`;
